src/differentials/thunks.jl

```diff
@@ -9,7 +9,9 @@ Base.Broadcast.broadcastable(x::AbstractThunk) = broadcastable(unthunk(x))
 end
 
 @inline function Base.iterate(::AbstractThunk, (val, state))
-    element, new_state = iterate(val, state)
+    next = iterate(val, state)
+    next isa Nothing && return nothing
```

Comment:
Doesn't really matter, but this is generally more idiomatic:
```suggestion
    next === nothing && return nothing
```